configure.ac

```diff
@@ -63,16 +63,48 @@ AC_ARG_ENABLE([esapi],
                             [build the esapi layer (default is yes)])],
             [enable_esapi=$enableval],
             [enable_esapi=yes])
-AS_IF([test "x$enable_esapi" != xno],
+
+
+AC_ARG_ENABLE([esapi], 
+    AS_HELP_STRING([--disable-esapi], [Enable TSS ESYS API library @<:@default=yes@:>@]),
+    [],
+    [enable_esapi="yes"])
+AM_CONDITIONAL(ESAPI, test "x$enable_esapi" = "xyes")
+
+AC_ARG_WITH([crypto],
+            [AS_HELP_STRING([--with-crypto={gcrypt,ossl}],
+                            [sets the ESAPI crypto backend (default is gcrypt)])],
+            [],
+            [with_crypto=gcrypt])
+
+AM_CONDITIONAL(ESYS_OSSL, test "x$with_crypto" = "xossl")
+AM_CONDITIONAL(ESYS_GCRYPT, test "x$with_crypto" = "xgcrypt")
+
+AS_IF([test "x$with_crypto" != "xgcrypt"],
+    AS_IF([test "x$with_crypto" != "xossl"],
+          AC_MSG_ERROR([Bad value for --with-crypto $with_crypto])))
+
+AS_IF([test "x$enable_esapi" != xno -a "x$with_cryptoprovider" = "xgcrypt"],
```

Comment:
$with_cryptoprovider? should be $with_crypto

Reply:
You are correct. Updated.